lms/static/js/components/card/views/card.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * A generic card view class.
+ *
+ * Subclasses can override any of the following:
+ * - configuration (string or function): Sets the display style of the card as square or list. Can take values of
+ *      "square_card" or "list_card". Defaults to "square_card".
+ * - action (function): Action to take when the action button is clicked. Defaults to a no-op.
+ * - cardClass (string or function): Class name for this card's DOM element. Defaults to the empty string.
+ * - title (string or function): Title of the card. Defaults to the empty string.
+ * - description (string or function): Description of the card. Defaults to the empty string.
+ * - details (array or function): Array of child views to be rendered as details of this card. The class "meta-detail"
+ *      is automatically added to each rendered child view to ensure appropriate styling. Defaults to an empty list.
+ * - actionClass (string or function): Class name for the action DOM element. Defaults to the empty string.
+ * - actionUrl (string or function): URL to navigate to when the action button is clicked. Defaults to the empty string.
+ * - actionContent: Content of the action button. This may include HTML. Default to the empty string.
+ */
+;(function (define) {
+    'use strict';
+    define(['jquery',
+            'backbone',
+            'text!templates/components/card/square_card.underscore',
+            'text!templates/components/card/list_card.underscore'],
+        function ($, Backbone, squareCardTemplate, listCardTemplate) {
+            var CardView = Backbone.View.extend({
+                events: {
+                    'click .action' : 'action'
+                },
+
+                switchOnConfiguration: function (square_result, list_result) {
+                    return (this.callIfFunction(this.configuration) || 'square_card') === 'square_card' ?
+                        square_result : list_result;
+                },
+
+                callIfFunction: function (value) {
+                    if ($.isFunction(value)) {
+                        return value.call(this);
+                    } else {
+                        return value;
+                    }
+                },
+
+                initialize: function () {
+                    this.template = this.switchOnConfiguration(
+                        _.template(squareCardTemplate),
+                        _.template(listCardTemplate)
+                    );
+                    this.render();
+                },
+
+                className: function () {
+                    return 'card ' +
+                        this.switchOnConfiguration('square-card', 'list-card') +
+                        ' ' + this.callIfFunction(this.cardClass);
+                },
+
+                configuration: function () {
+                    return this.options.configuration;
```

Comment:
I'm not crazy about relying on `this.options`, since it apparently goes away in the incremental version of Backbone released after the version we use. Since `"square_card"` is the default, you could define it in `initialize` like this:

```javascript
initialize: function (options) {
  ...
  this.configuration = options.configuration || "square_card";
  ...
},
```